src/Core/Authorization/AuthorizationResolver.cs

```diff
@@ -438,61 +440,150 @@ public IEnumerable<string> GetAllowedExposedColumns(string entityName, string ro
     }
 
     /// <summary>
-    /// Helper method to extract all claims available in the HttpContext's user object and add the claims
-    /// to the claimsInRequestContext dictionary to be used for claimType -> claim lookups.
+    /// Returns a dictionary (string, string) where a key is a claim's name and a value is a claim's value.
+    /// Resolves multiple claim objects of the same claim type into a JSON array mirroring the format
+    /// of the claim in the original JWT token. The JSON array's type depends on the value type
+    /// present in the original JWT token.
     /// </summary>
+    /// <remarks>
+    /// DotNet will resolve a claim with value type JSON array to multiple Claim objects
+    /// with the same type and different values.
+    /// e.g. roles and groups claim, which are arrays of strings and are flattened to:
+    /// roles: role1, roles: role2, groups: group1, groups: group2
+    /// "Claims are name valued pairs and nothing more."
+    /// Ref: https://github.com/dotnet/aspnetcore/issues/13647#issuecomment-527523224
+    /// The library that parses the JWT token into claims is the one that decides
+    /// *how* to resolve the claims from the token's JSON payload.
+    /// dotnet flattens the claims into a list:
+    /// https://github.com/dotnet/aspnetcore/blob/282bfc1b486ae235a3395150a8d53073a57b7f43/src/Security/Authentication/OAuth/src/JsonKeyClaimAction.cs#L39-L53
+    /// </remarks>
+    /// <param name="context">HttpContext which contains a ClaimsPrincipal</param>
+    /// <returns>Processed claims and claim values.</returns>
+    public static Dictionary<string, string> GetProcessedUserClaims(HttpContext? context)
+    {
+        Dictionary<string, string> processedClaims = new();
+
+        if (context is null)
+        {
+            return processedClaims;
+        }
+
+        Dictionary<string, List<Claim>> userClaims = GetAllAuthenticatedUserClaims(context);
+
+        foreach ((string claimName, List<Claim> claimValues) in userClaims)
+        {
+            // Some identity providers (other than Entra ID) may emit a 'scope' claim as a JSON string array. dotnet will
+            // create claim objects for each value in the array. DAB will honor that format
+            // and processes the 'scope' claim objects as a JSON array serialized to a string.
+            if (claimValues.Count > 1)
+            {
+                switch (claimValues.First().ValueType)
```

Comment:
nit: Is it possible for the `claimValues` to be an empty list here? In that, `.First()` would throw an exception

Reply:
No, For context we're talking about this:

```csharp
Dictionary<string, List<Claim>> userClaims = GetAllAuthenticatedUserClaims(context);

foreach ((string claimName, List<Claim> claimValues) in userClaims)
```

`userClaims` will not have an entry for a claim that doesn't exist. If there is an entry in the dictionary, there is at least one entry in the list of claims.

Reply:
also, the following `First()` is only used when `Count()` is > 1.
```csharp
if (claimValues.Count > 1)
{
    switch (claimValues.First().ValueType)
```

Reply:
My bad, I meant to add this in the `else` block. There `0`th element is accessed directly, hence I had the question. But, if there will be atleast one element in the `claimValues`, then it should be good.